src/components/shared/Settings.js

```diff
@@ -38,21 +30,22 @@ module.exports = React.createClass({
   },
 
   render: function() {
-    var editor = (store.get('editor')) ? store.get('editor') : 'ideditor';
+    const { editor, showSettings, closeSettings } = this.props;
+    console.log("SETTINGS", showSettings);
+    if (!showSettings) return null;
 
     return (
-      /* jshint ignore:start */
-      <div id='modal' className='animate modal modal-content active' onClick={this.onCancel}>
+      <div id='modal' className='animate modal modal-content active' onClick={closeSettings}>
```

Comment:
@ajithranka is it a good idea to use an id like `modal` here if it is only for this specific modal? Seems like we could run into conflicting ids on the page?